src/game/boe.town.cpp

```diff
@@ -1247,52 +1245,79 @@ void erase_specials() {
 }
 
 void erase_out_specials() {
-	
-	short out_num;
-	cSpecial sn;
-	short sd1,sd2;
-	location where;
-	
-	for(short i = 0; i < 2; i++)
-		for(short j = 0; j < 2; j++)
-			if(quadrant_legal(i,j)) {
-				cOutdoors& sector = *univ.scenario.outdoors[univ.party.outdoor_corner.x + i][univ.party.outdoor_corner.y + j];
-				for(short k = 0; k < sector.city_locs.size(); k++) {
-					if(sector.city_locs[k].spec >= 0 && sector.city_locs[k].x < 48 &&
-						univ.scenario.ter_types[sector.terrain[sector.city_locs[k].x][sector.city_locs[k].y]].special == eTerSpec::TOWN_ENTRANCE &&
-					   (sector.city_locs[k].x == minmax(0,47,sector.city_locs[k].x)) &&
-					   (sector.city_locs[k].y == minmax(0,47,sector.city_locs[k].y))) {
-						if(sector.city_locs[k].spec < 0 || sector.city_locs[k].spec >= univ.scenario.towns.size())
-							continue;
-						if(!univ.scenario.towns[sector.city_locs[k].spec]->can_find) {
-							univ.out[48 * i + sector.city_locs[k].x][48 * j + sector.city_locs[k].y] =
-								univ.scenario.ter_types[sector.terrain[sector.city_locs[k].x][sector.city_locs[k].y]].flag1;
-						}
-						else if(univ.scenario.towns[sector.city_locs[k].spec]->can_find) {
-							univ.out[48 * i + sector.city_locs[k].x][48 * j + sector.city_locs[k].y] =
-								sector.terrain[sector.city_locs[k].x][sector.city_locs[k].y];
-							
-						}
-					}
-				}
-				for(int k = 0; k < sector.special_locs.size(); k++) {
-					if(sector.special_locs[k].spec < 0) continue; // TODO: Is this needed? Seems to be important, so why was it commented out?
-					out_num = univ.scenario.outdoors.width() * (univ.party.outdoor_corner.y + j) + univ.party.outdoor_corner.x + i;
-					
-					sn = sector.specials[sector.special_locs[k].spec];
-					sd1 = sn.sd1; sd2 = sn.sd2;
-					if((univ.party.sd_legit(sd1,sd2)) && (PSD[sd1][sd2] == 250)) {
-						where = sector.special_locs[k];
-						if(where.x > 48 || where.y > 48 || where.x < 0 || where.y < 0) {
-							beep();
-							add_string_to_buf("Outdoor section corrupt. Problem fixed.");
-							sector.special_locs[k].spec = -1; // TODO: Again, was there a reason for commenting this out?
-						}
-						
-						sector.special_spot[where.x][where.y] = false;
-					}
-				}
+	for(short i = 0; i < 2; i++) {
+		for (short j = 0; j < 2; j++) {
+			if (!quadrant_legal(i, j)) continue;
+			auto& sector = get_sector_ptr(univ.party.outdoor_corner.x + i, univ.party.outdoor_corner.y + j);
+
+			erase_hidden_towns(sector, i, j);
+
+			erase_completed_specials(sector);
+		}
+	}
+}
+
+void erase_hidden_towns(cOutdoors& sector, int quadrant_x, int quadrant_y)
+{
+	for (short tile_index = 0; tile_index < sector.city_locs.size(); tile_index++) {
+		if (!sector.is_city_on_map(tile_index) ||
+			!is_tile_of_type(sector, tile_index, eTerSpec::TOWN_ENTRANCE) ||
+			!sector.is_tile_on_edge(tile_index)) {
+			continue;
+		}
+		if (!town_exists(sector, tile_index)) { continue; }
```

Comment:
If you want a one-liner if statement, please don't use braces. If you absolutely must use braces, please break it into three lines.

(My personal preference is the former, for the record.)